AsaApi/main.cpp

```diff
@@ -63,6 +63,67 @@ void OpenConsole()
 	SetConsoleOutputCP(CP_UTF8);
 }
 
+void DeleteOldLogFiles()
+{
+	const std::string config_path = AsaApi::Tools::GetCurrentDir() + "/config.json";
+	std::ifstream file{ config_path };
+	if (!file.is_open())
+		return;
+
+	nlohmann::json config;
+	file >> config;
+	file.close();
+
+	const std::string folderPath = API::Tools::GetCurrentDir() + "\\logs";
+	const int maxFiles = config["settings"].value("MaxFiles", 20); // 20 files
+	const int maxAgeInSeconds = config["settings"].value("MaxAgeInSeconds", 86400); //24hrs
+	const bool enableDeleteLogging = config["settings"].value("EnableDeleteLogging", true);
+
+	std::vector<std::pair<std::string, time_t>> fileTimePairs;
+	// Iterate through the files in the folder
+	for (const auto& entry : std::filesystem::directory_iterator(folderPath)) {
+		if (entry.is_regular_file()) {
+			std::string fileName = entry.path().filename().string();
+
+			// Extract datetime from the file name
+			std::string dateTimeString = fileName.substr(14, 19); // Extract data and time from "ArkApi_00000_yyyy-mm-dd_hh-mm.log"
```

Comment:
I'm not a fan of this, we should probably consider the actual system date/time of the file.

Reply:
i found a bug using substr as the process id could be 2 digits i will change it maybe splitting those underscore to get the date/time and i will look into reading the actual write date of the file